apis/Google.Cloud.Spanner.Data/Google.Cloud.Spanner.Data/SpannerCommand.ExecutableCommand.cs

```diff
@@ -106,9 +106,12 @@ internal async Task<SpannerDataReader> ExecuteReaderAsync(CommandBehavior behavi
                 ValidateConnectionAndCommandTextBuilder();
                 ValidateCommandBehavior(behavior);
 
+                if (CommandTextBuilder.SpannerCommandType == SpannerCommandType.Dml)
```

Comment:
Please use braces for all `if` statement bodies, even when there's only a single statement.
(I'm surprised our editorconfig didn't pick this up...)

Reply:
We're ignoring `behavior` and `singleUseReadSettings` here. Is that appropriate?

Reply:
(If they're not supported, we should validate that they're not set.)

Reply:
> So may be we can log a warning for it if for DML behaviour value is other than default, rather than validating null value, what do you think ?

No, warnings are very unlikely to be seen (the only logger we have easily available is the Spanner semi-internal one.

If the user has specified a behavior other than the default, and we're ignoring that, we should throw an exception to indicate that that behavior isn't supported. (I don't think it can actually be null.)

Reply:
Sure.

Reply:
Going through the existing code I believe that only `CommandBehavior.CloseConnection` is used in `ExecuteReader`  and that to decide that the connection is needed to be closed while disposing `SpannerDataReader` or not. This does not seems relevant with the current DML Implementations. Please correct me if I'm wrong.

Also in ExecuteReader here https://github.com/googleapis/google-cloud-dotnet/blob/799d9d1e2aff59c73c070fd8ceaecc89ade4d7ea/apis/Google.Cloud.Spanner.Data/Google.Cloud.Spanner.Data/SpannerCommand.ExecutableCommand.cs#L480 we already have a validation that  CommandBehaviour SchemaOnly is not supported by Cloud Spanner .

Reply:
Someone can call `ExecuteReader` and pass in `CommandBehavior.CloseConnection`, expecting (reasonably) that the connection should be closed. If we don't want to support it, then we should throw an exception if a client *tries* to do it - just ignoring their requested behavior seems wrong.

Reply:
Yes, we are throwing exception for it for DML at https://github.com/googleapis/google-cloud-dotnet/blob/579173ce98c110b54e354154be8e8e2311d6abc2/apis/Google.Cloud.Spanner.Data/Google.Cloud.Spanner.Data/SpannerCommand.ExecutableCommand.cs#L124

Reply:
Great - that wasn't there last time I'd reviewed :)